src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,64 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { useState } from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  onFilter: (filterBy: string, searchBy: string) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<Props> = ({ onFilter }) => {
+  const [status, setStatus] = useState('all');
+  const [searchBy, setSearchBy] = useState('');
+
+  const handleFilterSetting = (target: string) => {
+    setStatus(target || 'all');
+    onFilter(target, searchBy);
+  };
```

Comment:
and you don't need this function, just pass functions for query and status from `App` and filter it there